components/org.wso2.carbon.identity.application.authenticator.cognito/src/main/java/org/wso2/carbon/identity/application/authenticator/cognito/CognitoOIDCAuthenticator.java

```diff
@@ -0,0 +1,319 @@
+        logoutRedirectUrl.setRequired(true);
+        logoutRedirectUrl.setDescription("Enter logout redirect url");
+        logoutRedirectUrl.setDisplayOrder(5);
+        configProperties.add(logoutRedirectUrl);
+
+        Property scope = new Property();
+        scope.setDisplayName("Additional Query Parameters");
+        scope.setName(CognitoOIDCAuthenticatorConstants.ADDITIONAL_QUERY_PARAMS);
+        scope.setDescription("Additional query parameters. e.g: paramName1=value1");
+        scope.setDisplayOrder(6);
+        configProperties.add(scope);
+
+        return configProperties;
+    }
+
+    @Override
+    protected void initiateLogoutRequest(HttpServletRequest request, HttpServletResponse response,
+                                         AuthenticationContext context) throws LogoutFailedException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Sending logout request to external IDP");
+        }
+        String clientId = context.getAuthenticatorProperties().get(CognitoOIDCAuthenticatorConstants.CLIENT_ID);
+        String logoutUrl = getLogoutUrl(context.getAuthenticatorProperties());
+        String logoutRedirectUrl = context.getAuthenticatorProperties().get(CognitoOIDCAuthenticatorConstants
+                .LOGOUT_REDIRECT_URL);
+
+        Map<String, String> parameters = new HashMap<>();
+
+        parameters.put(CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT_CLIENT_ID, clientId);
+        parameters.put(CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT_URI, logoutRedirectUrl);
+        FrameworkUtils.setCookie(request, response, CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT_STATE_COOKIE,
+                context.getContextIdentifier(), null);
+        try {
+            logoutUrl = FrameworkUtils.appendQueryParamsStringToUrl(logoutUrl, getParamsString(parameters));
+            response.sendRedirect(logoutUrl);
+        } catch (IOException e) {
+            throw new LogoutFailedException("Error while triggering cognito logout for  " + clientId, e);
+        }
+
+    }
+
+    @Override
+    protected void processLogoutResponse(HttpServletRequest request, HttpServletResponse response,
+                                         AuthenticationContext context) throws LogoutFailedException {
+
+        FrameworkUtils.removeCookie(request, response, CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT_STATE_COOKIE);
+    }
+
+    private String getParamsString(Map<String, String> params) throws LogoutFailedException {
+
+        StringBuilder result = new StringBuilder();
+
+        try {
+            for (Map.Entry<String, String> entry : params.entrySet()) {
+                result.append(URLEncoder.encode(entry.getKey(), "UTF-8"));
+                result.append("=");
+                result.append(URLEncoder.encode(entry.getValue(), "UTF-8"));
+                result.append("&");
+            }
+
+            String resultString = result.toString();
```

Comment:
try catch is only required for for loop. so the return can be taken out of it which will make things compact and clean